cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -935,20 +947,29 @@ std::pair<std::vector<AtomKey>, std::vector<FrameAndDescriptor>> LocalVersionedE
     return std::make_pair(keys, folly::collect(results_fut).get());
 }
 
+folly::Future<std::pair<VersionedItem, FrameAndDescriptor>> LocalVersionedEngine::read_dataframe_version_get_future(
+        folly::Future<std::optional<AtomKey>>&& version_fut, const StreamId &stream_id, ReadQuery& read_query, const ReadOptions& read_options){
+    return  std::move(version_fut).via(&async::cpu_executor())
+    .thenValue([that = this, &stream_id, &read_query, &read_options](std::optional<AtomKey>&& key){
+        auto version = key.has_value() ? std::make_optional<VersionedItem>(std::move(to_atom(key.value()))) : std::nullopt;
+        return that->read_dataframe_and_validate_version_internal(version, stream_id, read_query, read_options);
+    }); 
+}
+
 std::vector<std::pair<VersionedItem, FrameAndDescriptor>> LocalVersionedEngine::batch_read_internal(
     const std::vector<StreamId>& stream_ids,
     const std::vector<VersionQuery>& version_queries,
     std::vector<ReadQuery>& read_queries,
-    const ReadOptions& read_options) {
-
+    const ReadOptions& read_options
+) {
+    std::vector<VersionQuery> empty_version_queries(stream_ids.size(), VersionQuery{});
```

Comment:
No. The way batch_get_versions is constructed, needs that the version vector to be the same size that the streams vector. but I agree that it can be create on-the-fly in the line just below. Amended that